test/NNPDE_tests_het.jl

```diff
@@ -0,0 +1,113 @@
+using Flux
+println("NNPDE_tests")
+using DiffEqFlux
+println("Starting Soon!")
+using ModelingToolkit
+using DiffEqBase
+using Test, NeuralPDE
+println("Starting Soon!")
+using GalacticOptim
+using Optim
+using Quadrature,Cubature, Cuba
+using QuasiMonteCarlo
+using SciMLBase
+using OrdinaryDiffEq
+import ModelingToolkit: Interval, infimum, supremum
+
+using Random
+Random.seed!(100)
+
+cb = function (p, l)
+    println("Current loss is: $l")
+    return false
+end
+
+## Example 1, heterogeneous equation
+function test_heterogeneous_equation(strategy_)
+	println("Example 1, Simple Heterogeneous input PDE, strategy: $strategy_")
+	@parameters x y
+	@variables p(..) q(..) r(..) s(..)
+	Dx = Differential(x)
+	Dy = Differential(y)
+
+	# 2D PDE
+	eq  = p(x) + q(y) + Dx(r(x, y)) + Dy(s(y, x)) ~ 0
+	# eq  = Dx(p(x)) + Dy(q(y)) + Dx(r(x, y)) + Dy(s(y, x)) + p(x) + q(y) + r(x, y) + s(y, x) ~ 0
+
+	# Initial and boundary conditions
+	bcs = [p(1) ~ 0.f0, q(-1) ~ 0.0f0,
+			r(x, -1) ~ 0.f0, r(1, y) ~ 0.0f0, 
+			s(y, 1) ~ 0.0f0, s(-1, x) ~ 0.0f0]
+	# bcs = [s(y, 1) ~ 0.0f0]
+	# Space and time domains
+	domains = [x ∈ IntervalDomain(0.0, 1.0),
+				y ∈ IntervalDomain(0.0, 1.0)]
+
+	# chain_ = FastChain(FastDense(2,12,Flux.σ),FastDense(12,12,Flux.σ),FastDense(12,1))
+	numhid = 3
+	fastchains = [[FastChain(FastDense(1, numhid, Flux.σ), FastDense(numhid, numhid, Flux.σ), FastDense(numhid, 1)) for i in 1:2];
+				[FastChain(FastDense(2, numhid, Flux.σ), FastDense(numhid, numhid, Flux.σ), FastDense(numhid, 1)) for i in 1:2]]
+	discretization = NeuralPDE.PhysicsInformedNN(fastchains,
+													strategy_)
+
+	pde_system = PDESystem(eq, bcs, domains, [x,y], [p(x), q(y), r(x, y), s(y, x)])
+	prob = SciMLBase.discretize(pde_system, discretization)
+	res = GalacticOptim.solve(prob, BFGS(); cb=cb, maxiters=2)
```

Comment:
it needs a convergence test with a heterogeneous input example.

Reply:
Totally. Do you know any good heterogeneous input system I could use to test?

Reply:
@ChrisRackauckas @KirillZubov how does one go about doing a convergence test when no analytic solution is available? I think I can solve an ODE in the case of the Kermack McKendrick theory but I am not sure how to square the discrete sol array with the continuous result of the PINN.

Also maybe my computer is slow but the Kermack McKendrick theory takes about 25 min. If you have any tips of a simpler system we can use for this convergence test, I'd appreciate it (for example, Zoe mentioned something about an SPM model in her PR). Otherwise I will just keep on refining this.

Reply:
The SPM one would be a hard one.